src/sql/parts/connection/connectionDialog/connectionWidget.ts

```diff
@@ -238,6 +241,31 @@ export class ConnectionWidget {
 		this._toDispose.push(this._passwordInputBox.onDidChange(passwordInput => {
 			this._password = passwordInput;
 		}));
+
+		this._toDispose.push(this._databaseNameInputBox.onFocus(() => {
+			if (this.serverName) {
```

Comment:
It might be best to do this only when the database list isn't already loaded. Right now it appears to reload every time I click on the database selection - not too bad connecting to my local server where the list comes back instantly, but could be annoying on a slow connection

Reply:
Added a cache to eliminate this problem